app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -232,7 +232,24 @@ public function fetchTransactionInfo(Mage_Payment_Model_Info $payment, $transact
     public function isAvailable($quote = null)
     {
         if(!empty($quote)) {
-            return $this->boltHelper()->canUseBolt($quote);
+            $canUseBolt = $this->boltHelper()->canUseBolt($quote);
+            $hidePayment = false;
+
+            $routeName= Mage::app()->getRequest()->getRouteName();
+            $moduleName = Mage::app()->getRequest()->getControllerModule();
+            $controllerName = Mage::app()->getRequest()->getControllerName();
+            $otherRoute = false;
+
+            Mage::dispatchEvent('bolt_is_available_payment', array('other_route' => $otherRoute));
```

Comment:
Here are some things to consider:

1.) It is not logical or normal to have a non-mutating event in a boolean getter method
2.) Even if such an event could be justified, the naming convention is incorrect.  It should begin with `bolt_boltpay_`
3.) A hard-code `false` parameter is of no use.  It appears that you are expecting to mutate it based on the code below.  However, you have only passed a copy of the the value of `$otherRoute` to your array.  You will therefore find that the value of $otherRoute in your if statement below, this will always be false.  Use our `doFilterEvent` construct instead.

Reply:
agree. updated.